src/driver/op/slab_list.rs

```diff
@@ -119,3 +120,53 @@ impl<'a, T> Drop for SlabList<'a, T> {
         }
     }
 }
+
+
+
+mod test {
+    use super::*;
+
+    #[test]
+    fn push_pop() {
+        let mut slab = Slab::with_capacity(8);
+        let mut list =  SlabListIndices::new().into_list(&mut slab);
+        assert!(list.is_empty());
+        assert_eq!(list.pop(), None);
+        for i in 0..5 {
+            list.push(i);
+            assert_eq!(list.peek_end(), Some(&i));
+            assert!(!list.is_empty());
+            assert!(!list.slab.is_empty());
+        }
+        for i in 0..5 {
+            assert_eq!(list.pop(), Some(i))
+        }
+        assert!(list.is_empty());
+        assert!(list.slab.is_empty());
+        assert_eq!(list.pop(), None);
+    }
+
+    #[test]
+    fn entries_freed_on_drop() {
+        let mut slab = Slab::with_capacity(8);
+        {
+            let mut list =  SlabListIndices::new().into_list(&mut slab);
+            list.push(42);
+            assert!(!list.is_empty());
+        }
+        assert!(slab.is_empty());
+    }
+
+    #[test]
+    fn entries_kept_on_converion_to_index() {
+        let mut slab = Slab::with_capacity(8);
+        {
+            let mut list =  SlabListIndices::new().into_list(&mut slab);
+            list.push(42);
+            assert!(!list.is_empty());
+            // This forgets the entries
+            let _ = list.into_indices();
+        }
+        assert!(!slab.is_empty());
+    }
+}
```

Comment:
Is this missing newline at end of file still a problem? I was trying to see why GitHub didn't think all of Noah's reviews were satisfied.